### DISCUSSION TITLE
Firmware upgrade procedure - in details

### DISCUSSION THREAD

**Discussion by ivdimitro** [zwave-js/zwave-js#1656] [Q&A] [answered]
Hi, I am thinking of updating my TriSensor firmware. The [Aeotec documentation](https://aeotec.freshdesk.com/support/solutions/articles/6000236093-how-to-update-trisensor-z-wave-firmware-) recommends to remove them from the network before upgrade. 
In zwave JS I can run the update directly as the device is in the network.
The questions is: 
- which procedure should I follow? 
- is the firmware update tested with real devices? 
- Anything else I need to know before I pull the trigger? 

Thanks. 

P.S. I see at least one report of successful upgrade so I should be safe right :)
P.S.1 Actually I just checked the Aeotec firmware and it is an exe file for windows only. Is there a way to scrape the binary or how to obtain the "naked" firmware?

**Comment by AlCalzone** [comment]
> which procedure should I follow?

Just do it!™
Jokes aside: Keep the device as close as possible to the controller, open zwavejs2mqtt dashboard, load your firmware update file, start.

> Is the firmware update tested with real devices?

Yes, I up- and downgraded my supporting devices (both aeotec) several times while developing the feature. From my experience the whole process is pretty safe and so far the devices never accepted corrupted firmware.

> Anything else I need to know before I pull the trigger?

Make sure you have the correct firmware. Using the wrong frequency (EU vs. US) could be bad. Obligatory: We don't take any responsibility.

**Comment by AlCalzone** [comment, marked as answer]
> P.S.1 Actually I just checked the Aeotec firmware and it is an exe file for windows only. Is there a way to scrape the binary or how to obtain the "naked" firmware?

Just put the exe into the update UI. zwave-js will try to extract the raw firmware.

**Reply by ivdimitro** [reply]
Damn I jumped the gun and updated the devices with the windows tool. I didn't expect you have implemented scraping the firmware from the exe file. I imagined that some hex editing will be involved and decided not to risk it. 
Good to know for the future and next time for sure is going via zwave JS. Thanks for the info.

**Reply by AlCalzone** [reply]
We err on the side of caution. If something seems off during firmware extraction, the process will fail.

**Comment by towerhand** [comment]
I have upgraded my zooz switches, it worked perfectly. Zjs2m will show the progress in the controller status box. 

I did have a couple show up as dead after the upgrade but just needed to heal or repair the node.
